CHANGELOG.md

```diff
@@ -1,35 +1,68 @@
 # textX changelog
 
-* Next Release 1.?.?
-   - https://github.com/textX/textX/pull/168
-     - Adding examples and documentation related to scope providers
-       (related to model modification through scope providers).
-   - https://github.com/textX/textX/pull/165
-     - Added metamodel export feature for PlantUML.
-   - https://github.com/textX/textX/pull/134
-     - Fixing bug when using a sequence of matches and rule reference in
-       an abstract rule choice alternative.
-     - Explicitly disallowing referencing more than one rule in an abstract
-       rule choice alternative.
-   - https://github.com/textX/textX/pull/114
-     - python like imports (named import rules, scope providers affected)
-     - OS incompatibility fixes (path separator).
-   - https://github.com/textX/textX/pull/98
-     - Added STRICTFLOAT as buildin type to allow to distinguish ints from
-       floats in NUMBERs. Fixed docu link.
-   - https://github.com/textX/textX/pull/93
-     - Changed attribute name for the metamodel object (from 
-       "metamodel._parser" to "metamodel._parser_blueprint").
-   - https://github.com/textX/textX/pull/120
-     Unicode requirement for (meta)-model strings API parameters made strict.
-     This should prevent common errors with Python 2.x
-     See:
-     https://github.com/textX/textX/issues/105
-     https://github.com/textX/textX/pull/99
-     https://github.com/textX/textX/issues/117
-
-
-* 2018-10-06 Release 1.8.0
+All _notable_ changes to this project will be documented in this file.
+
+The format is based on _[Keep a Changelog][keepachangelog]_, and this project
+adheres to _[Semantic Versioning][semver]_.
+
+Everything that is documented in the [official docs][textXDocs] is considered
```

Comment:
In principal this is OK for me. That is how I understand the responsibility of official docs. However, 'everything' is a hard word. Also all referenced unittests? With all details? 

We should at least exclude _underscore properties/functions, except the _tx_* aspects. This gives us the ability to mark implementation details in the future.

Reply:
Again, in principal it is OK for me! We should try hard to document all changes! So maybe ignore the last comment.

Reply:
Yup, we should somehow document the what should be considered a stable API for which change we shall increase the major version number. I guess the correct way is to document stuff that are user-facing and whenever we change those element is a backward incompatible way we do major number increase.

We should also make clear the policy we use to maintain old major versions. I guess we should give enough time for users to migrate to new major versions. Maybe to provide bugfixes to the next to last major version. E.g. when we release 2.0.0 we might provide bugfixes for 1.8.x until we hit 3.0.0. That should give plenty of time for the upgrade but will increase maintenance overhead. Not sure that we have a capacity for that.

Reply:
Even without support for previous version, users should be safe to declare their dependencies against the major version and receive all backward compatible improvements and bug fixes. So they can stay at previous version as long as they like. When the time is to upgrade to the next major version they should investigate the changelog for all breaking changes and see if they are affected. That should provide (I hope) smooth upgrades.

Reply:
OK for me.